draft-ietf-jsonpath-base.md

```diff
@@ -1036,28 +1036,30 @@ The current item is selected if and only if the result is `true`.
 
 
 ~~~~ abnf
+boolean-expr     = logical-or-expr / logical-and-expr
+logical-or-expr  = logical-and-expr "||" boolean-expr ; disjunction
+                                                      ; binds less tightly than conjunction
+logical-and-expr = basic-filter /
+                   basic-filter "&&" logical-and-expr ; conjunction
+                                                      ; binds more tightly than disjunction
 
```

Comment:
this is very clean and seems to manage binding preference for `and` ... great.

Reply:
Why are we moving away from the standard way of expressing grammars for languages with operator precedences?  Obviously, right recursion works, but we do have *(...) in ABNF and could use it.
(Right recursion leads to a binary tree that leans to the right, so a + b + c parses as a + (b + c).
Not a problem as long as operators are [associative](https://en.wikipedia.org/wiki/Operator_associativity).)
See the ECMAScript grammars for the usual way to do this.

Reply:
This was asked earlier and received little response.

Reply:
Where was what asked?

Reply:
@Glyn asked if there was a standard easy for ANBF to support operator precedence.

https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/110#issuecomment-887390352

You replied.

Reply:
> Why are we moving away from the standard way of expressing grammars for languages with operator precedences? 

Simply ignorance on my part.
> Obviously, right recursion works, but we do have *(...) in ABNF and could use it.
> (Right recursion leads to a binary tree that leans to the right, so a + b + c parses as a + (b + c).
> Not a problem as long as operators are [associative](https://en.wikipedia.org/wiki/Operator_associativity).)
> See the ECMAScript grammars for the usual way to do this.

I can't find an ABNF ECMAScript grammar. [This grammar](https://tc39.es/ecma262/#sec-multiplicative-operators) uses recursion, e.g.:
```
 AdditiveExpression :
AdditiveExpression + MultiplicativeExpression
AdditiveExpression - MultiplicativeExpression
```
@cabo Please could you provide a more specific reference to an ECMAScript grammar which demonstrates the "standard way" you are thinking of?

Reply:
@cabo I tried using repetition instead of right recursion. Please see what you make of it.

Reply:
Indeed, I must have misremembered the way TC39 writes their grammars, which apparently has morphed to a simple BNF (without repetition) plus weird two-level features (not quite Algol68 though :-).

Because different parser generators react in different ways to (immediate) recursion, I try to avoid that in favor of repetition.  I just looked at the Python grammar for a potential example, but that is a weird mix of the different styles.

31c1a7f is exactly what I had in mind.  Regular and non-surprising.  Thank you.

Reply:
Phew, thanks! :sweat_smile: